.github/workflows/cloud-tpu-ci-nightly.yml

```diff
@@ -0,0 +1,61 @@
+name: Cloud TPU nightly
+
+on:
+  schedule:
+    - cron: "0 14 * * *"  # daily at 7am PST
+  workflow_dispatch:  # allows triggering the workflow run manually
+
+# Should also be set to read-only in the project settings.
+permissions:
+  contents: read
+
+jobs:
+  cloud-tpu-test:
+    runs-on: [self-hosted, tpu, v4-8]
+    defaults:
+      run:
+        shell: bash -l {0}
+    strategy:
+      fail-fast: false  # don't cancel all jobs on failure
+      matrix:
+        python-version: ["3.10"]  # TODO(jakevdp): update to 3.11 when available.
+        jaxlib-version: [latest, nightly]
+    steps:
+      # https://opensource.google/documentation/reference/github/services#actions
+      # mandates using a specific commit for non-Google actions.
+      - uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8
```

Comment:
In our workflows we've added a comment like `# v3` to indicate what version this commit points to